qiskit_experiments/framework/backend_data.py

```diff
@@ -0,0 +1,210 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Backend data access helper class
+
+Since `BackendV1` and `BackendV2` do not share the same interface, this
+class unifies data access for various data fields.
+"""
+from qiskit.providers.models import PulseBackendConfiguration
+from qiskit.providers import BackendV1, BackendV2
+from qiskit.providers.fake_provider import fake_backend, FakeBackendV2, FakeBackend
+
+
+class BackendData:
+    """Class for providing joint interface for accessing backend data"""
+
+    def __init__(self, backend):
+        """Inits the backend and verifies version"""
+        self._backend = backend
```

Comment:
I like the idea of pulling out all the data in `__init__` so that the other methods can be common to V1 and V2. One question though -- would we need to use this compatibility class for any property of the backend that is allowed to be modified? Nothing in the class currently should be modified. `max_circuits` is an edge case which we have previously decided can be changed to help with job splitting: https://github.com/Qiskit/qiskit-experiments/issues/638#issuecomment-1035540914. However, modifying `max_experiments` like that comment describes is a hack according to this comment: https://github.com/Qiskit/qiskit-ibm-provider/issues/361#issuecomment-1158164167 (but it's the only way in qiskit-experiments to get a job split on a smaller number of circuits with BackendV1; with V2 the `max_circuits_per_job` run option can be used).

Besides `max_circuits`/`max_experiments`, is there anything else?

Reply:
I am very hesitant to accept the parsing solution. The backend comes from an outside source, and I don't know if and when it might be changed. I think of `BackendData` as a mediator class, which hides the differences between v1 and v2, not as something that somehow enables us to forget about the backend.

I considered using dispatcher but for now I actually prefer to keep the code as explicit as possible - I don't think it's large enough to drive us into using more abstract code.

Reply:
Fair enough, but I don't understand why
> I don't know if and when it might be changed

this becomes a problem for parsing. Actually backend is bit weird object because it is basically "configuration" + "interface for job submission". So current code allows us to write
```python
my_expr._backend_data._backend.run(...)
```
though it is not expected to call protected members.